src/EPR.Calculator.Frontend/wwwroot/js/Download.js

```diff
@@ -0,0 +1,35 @@
+﻿function downloadFile(url, errorAction, event) {
+    event.preventDefault();
+    $.ajax({
+        url: url,
+        method: 'GET',
+        xhrFields: {
+            responseType: 'blob'
+        },
+        success: function (data, status, xhr) {
+            // Get the Content-Disposition header
+            var contentDisposition = xhr.getResponseHeader('Content-Disposition');
```

Comment:
Even in success you might need to have a try/catch

`
   success: function (data, status, xhr) {
try {
  // Do the processing 
}
catch(err) {
  window.location.href = errorAction;
}
} // end of success
`